rma/models/rma.py

```diff
@@ -0,0 +1,904 @@
+
+    # CRUD methods (ORM overrides)
+    @api.model
+    def create(self, vals):
+        if vals.get('name', _('New')) == _('New'):
+            ir_sequence = self.env['ir.sequence']
+            if 'company_id' in vals:
+                ir_sequence = ir_sequence.with_context(
+                    force_company=vals['company_id'])
+            vals['name'] = ir_sequence.next_by_code('rma')
+        return super().create(vals)
+
+    # Action methods
+    @api.multi
+    def action_rma_send(self):
+        self.ensure_one()
+        template = self.env.ref('rma.mail_template_rma_notification', False)
+        form = self.env.ref('mail.email_compose_message_wizard_form', False)
+        ctx = {
+            'default_model': 'rma',
+            'default_res_id': self.ids[0],
+            'default_use_template': bool(template),
+            'default_template_id': template and template.id or False,
+            'default_composition_mode': 'comment',
+            'mark_rma_as_sent': True,
+            'model_description': 'RMA',
+            'force_email': True
+        }
+        return {
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'mail.compose.message',
+            'views': [(form.id, 'form')],
+            'view_id': form.id,
+            'target': 'new',
+            'context': ctx,
+        }
+
+    @api.multi
+    @api.returns('mail.message', lambda value: value.id)
+    def message_post(self, **kwargs):
+        """ Set 'sent' field to True when an email is sent from rma form
+        view. This field is used to set the appropriate style to the
+        'Send by Email' button in the rma form view.
+        """
+        if self.env.context.get('mark_rma_as_sent'):
+            self.write({'sent': True})
+        self_with_context = self.with_context(mail_post_autofollow=True)
+        return super(Rma, self_with_context).message_post(**kwargs)
+
+    @api.multi
+    def action_confirm(self):
+        """Invoked when 'Confirm' button in rma form view is clicked."""
+        self.ensure_one()
+        self._ensure_required_fields()
+        if self.state == 'draft':
+            if self.picking_id:
+                reception_move = self._create_receptions_from_picking()
+            else:
+                reception_move = self._create_receptions_from_product()
+            self.write({
+                'reception_move_id': reception_move.id,
+                'state': 'confirmed',
+            })
+
+    @api.multi
+    def action_refund(self):
+        """Invoked when 'Refund' button in rma form view is clicked
+        and 'rma_refund_action_server' server action is run.
+        """
+        refunds = {}
+        for record in self.filtered("can_be_refunded"):
+            refund = refunds.get(record.partner_invoice_id.id)
+            if not refund:
+                invoice_form = Form(self.env['account.invoice'].with_context(
```

Comment:
You know I don't like this way, but anyway, you could use a hook function like _prepare_invoice(record) to help override. Same for lines

Reply:
I'm not reluctant to it, I'm maybe too psycho-rigid to use something that has been written for tests in real code. Everyone has its defaults :smile:

Reply:
But you have seen that it's not anymore only for tests. It started for that, but can be used for more than that. Please don't stay just in the import.